package ru.demo.wms.exception;
/*
Класс ShipingNotFoundException является пользовательским исключением, наследующимся от RuntimeException. 
Это исключение предназначено для использования в ситуациях, когда информация о доставке (отправлении) 
не может быть найдена в системе. Это может произойти, например, при попытке доступа к данным об отправлении, 
которого нет в базе данных, или при обращении к информации о доставке, которая 
была удалена или еще не была введена в систему.

Исключение определяет два конструктора:

Без параметров: ShipingNotFoundException(), который вызывает базовый конструктор 
RuntimeException без сообщения. Этот конструктор может использоваться, когда конкретные 
детали ошибки не требуются.
С параметром сообщения: ShipingNotFoundException(String message), который также 
вызывает конструктор базового класса, но с предоставленным сообщением об ошибке. 
Это сообщение может быть использовано для передачи дополнительных деталей об ошибке, например, 
указания на конкретный идентификатор отправления, который не был найден.
Использование таких пользовательских исключений позволяет разработчикам более точно 
управлять потоком обработки ошибок в приложении и предоставлять более полезные сообщения об ошибках, 
улучшая тем самым пользовательский опыт и упрощая отладку и обслуживание кода.
*/
public class ShipingNotFoundException extends RuntimeException {

	private static final long serialVersionUID = 1L;

	public ShipingNotFoundException() {
		super();
	}

	public ShipingNotFoundException(String message) {
		super(message);
	}

}
